### PR TITLE
TMO_SHELL: DFU: library creation

### PR DESCRIPTION
**Pull request by kimmansfield**
First checkin of the code for a dfu gecko module

Signed-off-by: Kim Mansfield <kim.mansfield2@t-mobile.com>

### CODE REVIEW

**Comment by johnlange2** [comment]
dfu_gecko/prj.conf seems to include unnecessary configs. @kimmansfield can you please remove the ones that aren't needed?

File: samples/tmo_shell/src/tmo_dfu_download.h [hunk 1]
```diff
@@ -19,6 +19,7 @@ enum dfu_tgts {
 	DFU_9116W
 };
 
+#if 0
```
Comment (johnlange2) [inline samples/tmo_shell/src/tmo_dfu_download.h:22]: Can this 'dead' code just be removed?

File: samples/dfu_gecko/lib/dfu_gecko_lib.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#define DFU_URL_LEN  256
+
+struct dfu_file_t {
+	char desc[DFU_DESC_LEN];
+	char lfile[DFU_FILE_LEN];
+	char rfile[DFU_FILE_LEN];
+	char sha1[DFU_SHA1_LEN];
+};
+
+int dfu_mcu_firmware_upgrade(int slot_to_upgrade);
```
Comment (dchau-tmo) [inline samples/dfu_gecko/lib/dfu_gecko_lib.h:22]: @kimmansfield per our agreement from our discussion on 12/30/22 we wanted to define this API as so 
`int dfu_mcu_firmware_upgrade(int slot_to_upgrade, char *bin_file, char *sha_file);`

**Comment by dchau-tmo** [comment]
In the previous applications and including our samples we had use this definition 
`int dfu_mcu_get_oldest_slot();` Is this function not needed anymore. In the proposal we talked about exposing this as a API as its needed for FW updates.

File: samples/dfu_gecko/src/main.c [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+#include <dfu_gecko_lib.h>
+
+
+
+
+void main(void)
+
+{
+
+}
```
Comment (dchau-tmo) [inline samples/dfu_gecko/src/main.c:11]: This file is not doing anything. Can be removed? same with src folder.

**Review by jtbaumann — APPROVED** [review]

**Review by jtbaumann — APPROVED** [review]

**Review by johnlange2 — APPROVED** [review]

**Pull request merged**